project/src/main/java/com/google/sps/data/Event.java

```diff
@@ -249,4 +250,39 @@ public Builder toBuilder() {
         this.name, this.description, this.labels, this.location, this.date, this.time, this.host)
         .mergeFrom(this);
   }
+
+  /* Overridden because db always returns new Event obj
```

Comment:
Overriding equals is unrecommended.
https://docs.google.com/document/d/1Pxmkw6SooMGt5ah8HMyAVjtZ7vIvaEgGr8sLc7Efoo4/edit

Reply:
@nikhitashanker What was your final solution?

Reply:
It was to use assert equals and compare the fields individually. 
For example,
> Assert.assertEquals(actualOpportunity.getEventId(), EVENT_ID);
> Assert.assertEquals(actualOpportunity.getName(), NAME);
> Assert.assertEquals(actualOpportunity.getNumSpotsLeft(), NUMBER_OF_SPOTS);
The problem is when using test library functions like reflectionEquals if there is a field like Set<Volunteer> and the Volunteer equals method has not been overidden. Here, the library function uses the Set equals method which will compare the references of the volunteers rather than the attributes of the volunteer objects. If only Set<T> exists where T is a JAVA object like String then the reflectionEquals method can be used because String equals method is defined to compare contents of the strings rather than the references.

Reply:
Ok thank you!